proposals/bindless.md

```diff
@@ -397,46 +412,47 @@ Steps for `GPUBindGroup.update(binding, resource)`:
-   - Set the entry at `binding` in the bindgroup to `resource`.
+   - Set the entry at `slot` in the table to `resource`.
 
-Steps for `GPUBindGroup.insertBinding(resource)`:
+Steps for `GPUResourceTable.insertBinding(resource)`:
 
- - If `this.[[desc]].layout.[[desc]].dynamicArray` is `undefined`, throw an `OperationError`.
  - Let `slot` be `this.[[availableAfterSubmit]].findIndex((e) => e <= this.[[device]].queue.[[completedSubmitIndex]])`.
  - If `slot` is `undefined`, throw an `OperationError`.
- - Let `binding` be `slot + this.[[desc]].layout.[[desc]].dynamicArray.start`.
- - Call `this.update(binding, resource)`.
- - Return `binding`.
-
-Steps for `GPUBindGroup.removeBinding(binding)`:
+ - Call `this.update(slot, resource)`.
+ - Return `slot`.
 
+Steps for `GPUResourceTable.removeBinding(slot)`:
 
  - If any of the following is not satisfied, throw an `OperationError`:
 
-    - `this.[[desc]].layout.[[desc]].dynamicArray` must not be `undefined`.
-    - `binding` must be `>= this.[[desc]].layout.[[desc]].dynamic.start`.
-
- - Let `slot` be `binding - this.[[desc]].layout.[[desc]].dynamicArray.start`.
- - If any of the following is not satisfied, throw an `OperationError`:
-
-    - `this.destroy()` has never been called.
-    - `slot < Math.min(this.[[desc]].dynamicArraySize, this.[[device]].limits.maxDynamicBindingArraySize)`
+    - `this.destroy()` has never been called. (Note: this is a content timeline check, but `[[destroyed]]` is a device-timeline boolean)
+    - `slot` must be `< this.size`
 
  - Set `this.[[availableAfterSubmit]][slot]` to `this.[[device]].queue.[[lastSubmitIndex]]`.
 
  - On the device timeline:
 
    - If `this` isn't valid, generate a validation error and return.
-   - Remove the entry at `binding`.
+   - Remove the entry at `slot`.
 
-Note that the `Math.min()` with `maxDynamicBindingArraySize` is to allow browsers to skip allocating giant content-side arrays if a user tries sets `dynamicArraySize` to ludicrous numbers.
+### WGSL
 
-#### Shader/layout compatibility and default pipeline layout
+TODO: [#5380](https://github.com/gpuweb/gpuweb/issues/5380)
 
-TODO: [#5377](https://github.com/gpuweb/gpuweb/issues/5377)
+### Alternatives considered
 
-### WGSL
+#### Bindless GPUBindGroups
 
-TODO: [#5380](https://github.com/gpuweb/gpuweb/issues/5380)
+A previous version of the proposal added a "dynamic binding array" concept to `GPUBindGroup`, declared in the `GPUBindGroupLayoutDescriptor` with a starting binding and resource kind.
+The `GPUBindGroupDescriptor` had a creation argument that would decide of the size of the binding array and `GPUBindGroup` gained all the `destroy/update/insertBinding/removeBinding` methods that are on `GPUResourceTable`.
+In the shader the binding array could be accessed with `@group(N) @binding(M) var resources : resource_binding`.
+
+Discussion in [#5372](https://github.com/gpuweb/gpuweb/issues/5372) and offline determined that the `GPUResourceTable` approach was preferrable because:
```

Comment:
```suggestion
Discussion in [#5372](https://github.com/gpuweb/gpuweb/issues/5372) and offline determined that the `GPUResourceTable` approach was preferable because:
```